lib/api/model/events.dart

```diff
@@ -265,6 +267,89 @@ class MessageEvent extends Event {
   }
 }
 
+/// A Zulip event of type `update_message`.
+@JsonSerializable(fieldRename: FieldRename.snake)
+class UpdateMessageEvent extends Event {
+  @override
+  String get type => 'update_message';
+
+  final int? userId; // TODO(server-5)
+  final bool? renderingOnly; // TODO(server-5)
+  final int messageId;
+  final List<int> messageIds;
+  final List<String> flags;
+  final int? editTimestamp; // TODO(server-5)
+  final String? streamName;
+  final int? streamId;
+  final int? newStreamId;
+  final String? propagateMode;
+  final String? origSubject;
+  final String? subject;
+  final List<Object>? topicLinks; // TODO shape of objects
+  final String? origContent;
+  final String? origRenderedContent;
+  final int? prevRenderedContentVersion;
+  final String? content;
+  final String? renderedContent;
+  final bool? isMeMessage;
+
+  UpdateMessageEvent({
+    required super.id,
+    this.userId,
+    this.renderingOnly,
+    required this.messageId,
+    required this.messageIds,
+    required this.flags,
+    this.editTimestamp,
+    this.streamName,
+    this.streamId,
+    this.newStreamId,
+    this.propagateMode,
+    this.origSubject,
+    this.subject,
+    this.topicLinks,
+    this.origContent,
+    this.origRenderedContent,
+    this.prevRenderedContentVersion,
+    this.content,
+    this.renderedContent,
+    this.isMeMessage,
+  });
+
+  factory UpdateMessageEvent.fromJson(Map<String, dynamic> json) =>
+    _$UpdateMessageEventFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() => _$UpdateMessageEventToJson(this);
+}
+
+/// A Zulip event of type `delete_message`.
+@JsonSerializable(fieldRename: FieldRename.snake)
+class DeleteMessageEvent extends Event {
+  @override
+  String get type => 'delete_message';
+
+  final List<int> messageIds;
+  // final int messageId; // Not present; we support the bulk_message_deletion capability
+  final String messageType;
```

Comment:
This is really an enum in the API, not an arbitrary string, so it should be an enum here.

… Or given the behavior of `streamId` and `topic`, the most type-safe version would be to make `DeleteMessageEvent` abstract and have two subclasses `DeleteStreamMessageEvent` and `DeleteDmMessageEvent`, akin to `Message`/`StreamMessage`/`DmMessage`. But the amount of code that's going to care about these objects is pretty limited — should be a small amount of code for handling the event in various data structures, and then we discard the event — so sticking with one class is fine so long as that code comes out reasonably clean.

Reply:
Yeah. Also, before using `messageType`, `streamId`, or `topic` for anything, I'd want to better nail down what they actually mean, since (with `bulk_message_deletion `) we're given a list of message IDs, not a single message ID. Is the list guaranteed to be all stream messages or all DMs? If all stream messages, will they definitely all be in the same stream/topic?

For any given message in `message_ids`, it seems natural to just look up the message in our data structures and find out what its type is, and its stream/topic if relevant.

Reply:
> Is the list guaranteed to be all stream messages or all DMs? If all stream messages, will they definitely all be in the same stream/topic?

Pretty sure if those properties are present, they'll apply to all the messages in the event. (And I guess one of them is always present.) But it'd be good to nail that down.



> For any given message in `message_ids`, it seems natural to just look up the message in our data structures and find out what its type is, and its stream/topic if relevant, if we need that information.

Yeah. I'm not sure what the backstory was for why that other information is included here.